# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/find/nsWebBrowserFind.h
# Commit: e07e5ea2f75e
# Full Hash: e07e5ea2f75e971967decbbe8f42aff80987fda2
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-02-16 08:52:20
# Regressor Bug: 1945862
# File Overlap Count: 1
# Description:
#   Bug 1945862 - Make `nsFind` respect the end point correctly. r=emilio
#   
#   This patch refactors the find functionality in a few ways:
#   - A C++-only method `FindFromRangeBoundaries()` is introduced,
#     which uses `RangeBoundary`s instead of collapsed `nsRange`s
# ==============================================================================

diff -r 5a1f8c0bb78f -r e07e5ea2f75e toolkit/components/find/nsWebBrowserFind.h
--- a/toolkit/components/find/nsWebBrowserFind.h	Sat Feb 15 15:19:21 2025 +0000
+++ b/toolkit/components/find/nsWebBrowserFind.h	Sat Feb 15 16:02:34 2025 +0000
@@ -66,13 +66,11 @@
   MOZ_CAN_RUN_SCRIPT_BOUNDARY void SetSelectionAndScroll(
       nsPIDOMWindowOuter* aWindow, nsRange* aRange);
 
-  nsresult GetRootNode(mozilla::dom::Document* aDomDoc,
-                       mozilla::dom::Element** aNode);
-  nsresult GetSearchLimits(nsRange* aRange, nsRange* aStartPt, nsRange* aEndPt,
-                           mozilla::dom::Document* aDoc,
+  nsresult GetSearchLimits(nsRange* aSearchRange, nsRange* aStartPt,
+                           nsRange* aEndPt, mozilla::dom::Document* aDoc,
                            mozilla::dom::Selection* aSel, bool aWrap);
-  nsresult SetRangeAroundDocument(nsRange* aSearchRange, nsRange* aStartPoint,
-                                  nsRange* aEndPoint,
+  nsresult SetRangeAroundDocument(nsRange* aSearchRange, nsRange* aStartPt,
+                                  nsRange* aEndPt,
                                   mozilla::dom::Document* aDoc);
 
  protected: